var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -84,7 +87,7 @@ class Qgis(CMakePackage):
     depends_on('proj@4.4.0:')
     depends_on('py-psycopg2', type=('build', 'run'))  # TODO: is build dependency necessary?
     depends_on('py-pyqt4', when='@2')
-    depends_on('py-pyqt5@5.3:', when='@3')
+    depends_on('py-pyqt5@5.3: +qsci_api', when='@3')
```

Comment:
In the past, we avoided adding `+qsci_api` since this is not a requirement by QGIS.

Reply:
Could you instead do this in the section starting with `# Take care of conflicts ...` on line 128.

Reply:
I'm not sure to understand. You would like me to shift line 90 to line 128, is that it ?
Anyway, as you obviously master qgis, don't you think we could forget about my PR so that you get free to make your own, possibly or not with 3.16.4 ?
Indeed I modified qgis since I wanted to check my modifs to make qscintilla work for Octave (without python) did not degrade other packages. This is where I met this GDAL cmake error on CentOS 7. That's the complete story.